src/main/kotlin/no/nav/syfo/client/narmesteleder/NarmesteLederClient.kt

```diff
@@ -13,91 +11,77 @@ import no.nav.syfo.domain.Virksomhetsnummer
 import no.nav.syfo.util.*
 import org.slf4j.LoggerFactory
 
+data class NarmesteLederRelasjonDTO(
+    val narmesteLederRelasjon: NarmesteLederDTO
+)
+
 class NarmesteLederClient(
-    private val azureAdV2Client: AzureAdV2Client,
-    private val modiasyforestClientId: String,
-    modiasyforestBaseUrl: String
+    narmesteLederBaseUrl: String,
+    private val narmestelederClientId: String,
+    private val azureAdV2Client: AzureAdV2Client
 ) {
-    private val personNarmesteLederUrl: String
-    private val personNarmesteLederV2Url: String
-
-    init {
-        this.personNarmesteLederUrl = "$modiasyforestBaseUrl$PERSON_NARMESTELEDER_PATH"
-        this.personNarmesteLederV2Url = "$modiasyforestBaseUrl$PERSON_V2_NARMESTELEDER_PATH"
-    }
+    private val sykmeldtNarmesteLederePath = "$narmesteLederBaseUrl/sykmeldt/narmesteledere?utvidet=ja"
+    private val sykmeldtAktivNarmesteLederPath = "$narmesteLederBaseUrl/sykmeldt/narmesteleder?orgnummer="
 
     private val httpClient = httpClientDefault()
 
     suspend fun activeLeader(
         personIdentNumber: PersonIdentNumber,
         virksomhetsnummer: Virksomhetsnummer,
-        token: String,
-        callId: String,
-        onBehalfOf: Boolean = false,
+        callId: String
     ): NarmesteLederDTO? {
-        return narmesteLedere(
-            personIdentNumber = personIdentNumber,
-            token = token,
-            callId = callId,
-            onBehalfOf = onBehalfOf,
-        ).filter {
-            it.orgnummer == virksomhetsnummer.value
-        }.maxByOrNull {
-            it.fomDato
+
+        val systemToken = azureAdV2Client.getSystemToken(
+            scopeClientId = narmestelederClientId,
+        )?.accessToken
+            ?: throw RuntimeException("Failed to request access to Narmesteleder: Failed to get System token")
+
+        try {
+            val narmesteLederRelasjon: NarmesteLederRelasjonDTO =
+                httpClient.get("$sykmeldtAktivNarmesteLederPath${virksomhetsnummer.value}") {
+                    header(HttpHeaders.Authorization, bearerHeader(systemToken))
+                    header("Sykmeldt-Fnr", personIdentNumber.value)
+                    accept(ContentType.Application.Json)
+                }
+
+            return narmesteLederRelasjon.narmesteLederRelasjon
+        } catch (ex: ClientRequestException) {
+            log.error(
+                "Error while requesting NarmesteLedere of person from $sykmeldtAktivNarmesteLederPath${virksomhetsnummer.value} with {}, {}",
+                StructuredArguments.keyValue("statusCode", ex.response.status.value.toString()),
+                callIdArgument(callId)
+            )
+            return null
         }
     }
 
     suspend fun narmesteLedere(
         personIdentNumber: PersonIdentNumber,
-        token: String,
-        callId: String,
-        onBehalfOf: Boolean,
+        callId: String
```

Comment:
Pirk: Ikke glemt at trailing comma er støttet  i Kotlin nå og bør brukes for å gjøre det mer git-vennlig. Dette gjelder også mange andre steder i denne PR'en.